From 83b676bf136095f3069c219810ea361829d81558 Mon Sep 17 00:00:00 2001
From: Matthias Grimm <mgr@hobel.fritz.box>
Date: Fri, 20 Nov 2015 18:47:00 +0100
Subject: [PATCH] Added some more hue devices and Gamut-Profile

Added the new Lightstrip plus, hue go and Aura. The first two come with
a new Gamut profile and the Aura is from Living Colors.
---
 libsrc/leddevice/LedDevicePhilipsHue.cpp | 40 ++++++++++++++++++--------------
 1 file changed, 23 insertions(+), 17 deletions(-)

diff --git a/libsrc/leddevice/LedDevicePhilipsHue.cpp b/libsrc/leddevice/LedDevicePhilipsHue.cpp
index c65391f..dd58e12 100755
--- a/libsrc/leddevice/LedDevicePhilipsHue.cpp
+++ b/libsrc/leddevice/LedDevicePhilipsHue.cpp
@@ -20,23 +20,29 @@ bool operator !=(CiColor p1, CiColor p2) {
 	return !(p1 == p2);
 }
 
-PhilipsHueLight::PhilipsHueLight(unsigned int id, QString originalState, QString modelId) :
-		id(id), originalState(originalState) {
-	// Hue system model ids (http://www.developers.meethue.com/documentation/supported-lights).
-	// Light strips, color iris, ...
-	const std::set<QString> GAMUT_A_MODEL_IDS = { "LLC001", "LLC005", "LLC006", "LLC007", "LLC010", "LLC011", "LLC012",
-			"LLC013", "LST001" };
-	// Hue bulbs, spots, ...
-	const std::set<QString> GAMUT_B_MODEL_IDS = { "LCT001", "LCT002", "LCT003", "LLM001" };
-	// Find id in the sets and set the appropiate color space.
-	if (GAMUT_A_MODEL_IDS.find(modelId) != GAMUT_A_MODEL_IDS.end()) {
-		colorSpace.red = {0.703f, 0.296f};
-		colorSpace.green = {0.2151f, 0.7106f};
-		colorSpace.blue = {0.138f, 0.08f};
-	} else if (GAMUT_B_MODEL_IDS.find(modelId) != GAMUT_B_MODEL_IDS.end()) {
-		colorSpace.red = {0.675f, 0.322f};
-		colorSpace.green = {0.4091f, 0.518f};
-		colorSpace.blue = {0.167f, 0.04f};
+PhilipsHueLight::PhilipsHueLight(unsigned int id, QString originalState, QString modelId) :
+    id(id), originalState(originalState) {
+        // Hue system model ids (http://www.developers.meethue.com/documentation/supported-lights).
+        // Light strips, color iris, ...
+  const std::set<QString> GAMUT_A_MODEL_IDS = { "LLC001", "LLC005", "LLC006", "LLC007", "LLC010", "LLC011", "LLC012",
+                        "LLC013", "LLC014", "LST001" };
+  // Hue bulbs, spots, ...
+  const std::set<QString> GAMUT_B_MODEL_IDS = { "LCT001", "LCT002", "LCT003", "LCT007", "LLM001" };
+  // Hue Lightstrip plus, go ...
+  const std::set<QString> GAMUT_C_MODEL_IDS = { "LLC020", "LST002" };
+  // Find id in the sets and set the appropiate color space.
+      if (GAMUT_A_MODEL_IDS.find(modelId) != GAMUT_A_MODEL_IDS.end()) {
+              colorSpace.red = {0.703f, 0.296f};
+              colorSpace.green = {0.2151f, 0.7106f};
+              colorSpace.blue = {0.138f, 0.08f};
+      } else if (GAMUT_B_MODEL_IDS.find(modelId) != GAMUT_B_MODEL_IDS.end()) {
+              colorSpace.red = {0.675f, 0.322f};
+              colorSpace.green = {0.4091f, 0.518f};
+              colorSpace.blue = {0.167f, 0.04f};
+      } else if (GAMUT_C_MODEL_IDS.find(modelId) != GAMUT_B_MODEL_IDS.end()) {
+              colorSpace.red = {0.675f, 0.322f};
+              colorSpace.green = {0.2151f, 0.7106f};
+              colorSpace.blue = {0.167f, 0.04f};
 	} else {
 		colorSpace.red = {1.0f, 0.0f};
 		colorSpace.green = {0.0f, 1.0f};
